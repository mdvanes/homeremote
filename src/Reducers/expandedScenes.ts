import { TOGGLE_EXPAND_SCENE } from '../Actions/index';

type Action = {
    type: string;
    sceneIdx: string;
};

type State = {
    expanded: any[];
};

const expandedScenes = (state: State = { expanded: [] }, action: Action) => {
    if (action.type === TOGGLE_EXPAND_SCENE) {
        if (state.expanded.includes(action.sceneIdx)) {
            // Exists already, so toggle to "collapsed" and remove from the expanded array
            return {
                expanded: state.expanded.filter(id => id !== action.sceneIdx)
            };
        }
        // Does not exist yet, so toggle to "expanded" and add to the expanded array
        return {
            expanded: [...state.expanded, action.sceneIdx]
        };
    }
    return state;
};

export default expandedScenes;
