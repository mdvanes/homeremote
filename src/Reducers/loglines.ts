import {
    LOG_INFO,
    LOG_ERROR,
    CLEAR_LOG,
    LogInfoAction,
    LogErrorAction,
} from "../Actions";

type Logline = {
    message: string;
};

type State = Logline[];

// TODO this should be the union of all possible Actions in the application
type Actions = LogInfoAction | LogErrorAction;

// TODO use type State from LogContainer
type LoglinesReducer = (state: State, action: Actions) => State;

const loglines: LoglinesReducer = (state = [], action) => {
    // Too keep cyclomatic complexity low
    const newLogState: State = [
        ...state,
        {
            message: action.message,
        },
    ];
    const actionTypeMap: Record<string, State> = {
        [LOG_ERROR]: newLogState,
        [LOG_INFO]: newLogState,
        [CLEAR_LOG]: [],
    };
    if (actionTypeMap.hasOwnProperty(action.type)) {
        return actionTypeMap[action.type];
    }
    return state;
    // switch (action.type) {
    //     case LOG_ERROR:
    //     case LOG_INFO:
    //         return [
    //             ...state,
    //             {
    //                 message: action.message
    //             }
    //         ];
    //     case CLEAR_LOG:
    //         return [];
    //     default:
    //         return state;
    // }
};

export default loglines;
