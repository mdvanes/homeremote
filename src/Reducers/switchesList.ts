// import { SET_SWITCHES, SetSwitchesAction } from '../Actions';

// Domoticz Device Switch
export type DSwitch = {
    idx: string;
    name: string;
    type: string;
    status: string;
    dimLevel: number;
    readOnly: boolean;
    children: DSwitch[] | false;
};

type State = {
    switches: DSwitch[];
};

// const emptyList: DSwitch[] = [];

// type SwitchesListReducer = (state: State, action: SetSwitchesAction) => State;

// const switchesList: SwitchesListReducer = (
//     state = { switches: emptyList },
//     action
// ) => {
//     if (action.type === SET_SWITCHES) {
//         return {
//             switches: action.switches
//         };
//     }
//     return state;
// };

// export default switchesList;
