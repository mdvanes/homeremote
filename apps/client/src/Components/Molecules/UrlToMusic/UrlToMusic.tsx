import React, { FC } from "react";
import {
    Button,
    Card,
    CardContent,
    TextField,
    LinearProgress,
} from "@mui/material";
import Alert from "@material-ui/lab/Alert";
import { useDispatch, useSelector } from "react-redux";
import { RootState } from "../../../Reducers";
import {
    UrlToMusicState,
    getInfo,
    setFormField,
    setFormFieldError,
    getMusic,
} from "./urlToMusicSlice";

const FILL_IN_THIS_FIELD = "Fill in this field";

const UrlToMusic: FC = () => {
    const dispatch = useDispatch();

    const {
        error: errorMessage,
        form,
        isLoading,
        result,
    } = useSelector<RootState, UrlToMusicState>(
        (state: RootState) => state.urlToMusic
    );

    const validateGetInfo = (): boolean => {
        if (form.url.value === "") {
            dispatch(setFormFieldError(["url", FILL_IN_THIS_FIELD]));
            return false;
        }
        return true;
    };

    const handleUrlChange = (event: React.ChangeEvent<HTMLInputElement>) => {
        dispatch(setFormField([event.target.name, event.target.value]));
    };

    const dispatchGetInfo = () => {
        const isValid = validateGetInfo();
        if (isValid) {
            dispatch(getInfo());
        }
    };

    const handleMusicDataChange = (
        event: React.ChangeEvent<HTMLInputElement>
    ) => {
        dispatch(setFormField([event.target.name, event.target.value]));
    };

    const validateGetMusic = (): boolean => {
        const invalidFieldNames = Object.entries(form)
            .filter(([, formField]) => formField.value === "")
            .map(([fieldName]) => fieldName);
        invalidFieldNames.forEach((fieldName) => {
            dispatch(setFormFieldError([fieldName, FILL_IN_THIS_FIELD]));
        });
        return invalidFieldNames.length === 0;
    };

    const dispatchGetMusic = () => {
        const isValid = validateGetMusic();
        if (isValid) {
            dispatch(getMusic());
        }
    };

    return (
        <Card className="card-dashboard-height">
            <CardContent>
                <form>
                    <TextField
                        data-testid="url"
                        label="URL"
                        name="url"
                        fullWidth={true}
                        value={form.url.value}
                        error={Boolean(form.url.error)}
                        helperText={form.url.error}
                        onChange={handleUrlChange}
                    />
                    <Button
                        data-testid="get-info"
                        color="primary"
                        onClick={dispatchGetInfo}
                    >
                        Get Info
                    </Button>
                </form>
                <form>
                    <TextField
                        data-testid="title"
                        label="Title"
                        name="title"
                        fullWidth={true}
                        value={form.title.value}
                        error={Boolean(form.title.error)}
                        helperText={form.title.error}
                        onChange={handleMusicDataChange}
                    />
                    <TextField
                        data-testid="artist"
                        label="Artist"
                        name="artist"
                        fullWidth={true}
                        value={form.artist.value}
                        error={Boolean(form.artist.error)}
                        helperText={form.artist.error}
                        onChange={handleMusicDataChange}
                    />
                    <TextField
                        label="Album"
                        name="album"
                        fullWidth={true}
                        value={form.album.value}
                        error={Boolean(form.album.error)}
                        helperText={form.album.error}
                        onChange={handleMusicDataChange}
                    />
                    <Button
                        data-testid="get-music"
                        color="primary"
                        onClick={dispatchGetMusic}
                    >
                        Get Music
                    </Button>
                </form>
                {errorMessage && <Alert severity="error">{errorMessage}</Alert>}
                {result && <Alert severity="success">Result in {result}</Alert>}
                {isLoading && <LinearProgress />}
            </CardContent>
        </Card>
    );
};

export default UrlToMusic;
