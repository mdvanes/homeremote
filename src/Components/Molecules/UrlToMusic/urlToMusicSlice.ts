import { createAsyncThunk, createSlice, PayloadAction } from "@reduxjs/toolkit";
import fetchToJson from "../../../fetchToJson";
import { RootState } from "../../../Reducers";
import ApiBaseState from "../../../Reducers/state.types";
import { logInfo } from "../LogCard/logSlice";

interface FormField {
    value: string;
    error: string | false;
}

export interface UrlToMusicState extends ApiBaseState {
    form: Record<string, FormField>;
}

const year = new Date().getFullYear();

export const initialState: UrlToMusicState = {
    isLoading: false,
    error: false,
    form: {
        url: {
            value: "",
            error: false,
        },
        title: {
            value: "",
            error: false,
        },
        artist: {
            value: "",
            error: false,
        },
        album: {
            value: `Songs from ${year}`,
            error: false,
        },
    },
};

interface FetchInfoReturned {
    // url: string;
    title: string;
}

export const getInfo = createAsyncThunk<FetchInfoReturned>(
    `urlToMusic/getInfo`,
    async (_, { getState, dispatch }) => {
        const url = (getState() as RootState).urlToMusic.form.url.value;
        return fetchToJson<FetchInfoReturned>("/api/urltomusic/getinfo", {
            method: "POST",
            body: JSON.stringify({ url }),
        });
    }
);

export const getMusic = createAsyncThunk<FetchInfoReturned>(
    `urlToMusic/getMusic`,
    async (_, { getState, dispatch }) => {
        const form = (getState() as RootState).urlToMusic.form;
        const fieldEntries = Object.entries(form).map(([fieldName, field]) => [
            fieldName,
            field.value,
        ]);
        const body = JSON.stringify(Object.fromEntries(fieldEntries));
        dispatch(
            logInfo(
                `Started getMusic: ${fieldEntries
                    .map(([fieldName, value]) => `${fieldName}=${value}`)
                    .join("; ")}`
            )
        );
        return fetchToJson<FetchInfoReturned>("/api/urltomusic/getmusic", {
            method: "POST",
            body,
        });
    }
);

type NameAndValue = [string, string];

const urlToMusicSlice = createSlice({
    name: "urlToMusic",
    initialState,
    reducers: {
        setFormField: (
            draft,
            { payload }: PayloadAction<NameAndValue>
        ): void => {
            const [name, value] = payload;
            draft.form[name].value = value;
            draft.form[name].error = "";
        },
        setFormFieldError: (
            draft,
            { payload }: PayloadAction<NameAndValue>
        ): void => {
            const [name, error] = payload;
            draft.form[name].error = error;
        },
    },
    extraReducers: (builder) => {
        builder.addCase(getInfo.pending, (draft, { payload }): void => {
            draft.error = initialState.error;
            draft.isLoading = true;
        });
        builder.addCase(getInfo.fulfilled, (draft, { payload }): void => {
            draft.isLoading = initialState.isLoading;
            draft.form.title.value = payload.title; // TODO set all
        });
        builder.addCase(getInfo.rejected, (draft, { error }): void => {
            draft.isLoading = initialState.isLoading;
            draft.form.title.value = ""; // TODO clear all except url?
            draft.error = error.message || "An error occurred";
        });
        builder.addCase(getMusic.pending, (draft, { payload }): void => {
            draft.error = initialState.error;
            draft.isLoading = true;
        });
        builder.addCase(getMusic.fulfilled, (draft, { payload }): void => {
            draft.isLoading = initialState.isLoading;
            draft.form.title.value = payload.title;
        });
        builder.addCase(getMusic.rejected, (draft, { error }): void => {
            draft.isLoading = initialState.isLoading;
            draft.error = error.message || "An error occurred";
        });
    },
});

export const { setFormField, setFormFieldError } = urlToMusicSlice.actions;

export default urlToMusicSlice.reducer;
