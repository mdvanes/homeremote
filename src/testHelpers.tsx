import React, { FC, ReactElement } from "react";
import { render, RenderResult, RenderOptions } from "@testing-library/react";
import { Provider } from "react-redux";
import {
    combineReducers,
    configureStore,
    Reducer,
    Store,
} from "@reduxjs/toolkit";
import { RootState } from "./Reducers";

interface RenderWithProvidersOptions extends RenderOptions {
    initialState: Partial<RootState>;
    reducers: Record<string, Reducer>;
    store?: Store;
}

type RenderWithProviders = (
    ui: ReactElement,
    options: RenderWithProvidersOptions
) => RenderResult;

export const renderWithProviders: RenderWithProviders = (
    ui,
    {
        initialState,
        reducers,
        store = configureStore({
            reducer: combineReducers(reducers),
            preloadedState: initialState,
        }),
        ...renderOptions
    }
) => {
    const Wrapper: FC = ({ children }) => {
        return <Provider store={store}>{children}</Provider>;
    };
    return render(ui, { wrapper: Wrapper, ...renderOptions });
};

// For RTK Query
type MockStoreProviderProps = {
    api: any;
};

export const MockStoreProvider: FC<MockStoreProviderProps> = ({
    api,
    children,
}) => {
    const rootReducer = combineReducers({
        [api.reducerPath]: api.reducer,
    });

    const store = configureStore({
        reducer: rootReducer,
        middleware: (getDefaultMiddleware) =>
            getDefaultMiddleware().concat(api.middleware),
    });

    return <Provider store={store}>{children}</Provider>;
};
