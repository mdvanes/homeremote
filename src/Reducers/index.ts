import loglines from './loglines';
// import short from './short';
// import moveParams from './moveParams';
// import moveProgress from './moveProgress';
// import fileManager from './fileManager';
import switchesList from './switchesList';
import expandedScenes from './expandedScenes';
import { combineReducers } from 'redux';

const homeRemoteReducers = combineReducers({
    loglines,
    // short,
    // moveParams,
    // moveProgress,
    // fileManager,
    switchesList,
    expandedScenes
});

export default homeRemoteReducers;
