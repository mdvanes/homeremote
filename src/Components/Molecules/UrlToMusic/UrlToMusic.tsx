import { Button, Card, CardContent, TextField } from "@material-ui/core";
import React, { FC } from "react";
import { useDispatch, useSelector } from "react-redux";
import { RootState } from "../../../Reducers";
import { logInfo } from "../LogCard/logSlice";
import { UrlToMusicState, getInfo, setUrl } from "./urlToMusicSlice";

const UrlToMusic: FC = () => {
    const dispatch = useDispatch();

    const { url, title, error: errorMessage } = useSelector<
        RootState,
        UrlToMusicState
    >((state: RootState) => state.urlToMusic);
    // const title = useSelector<RootState, UrlToMusicState["title"]>(
    //     (state: RootState) => state.urlToMusic.title
    // );
    // TODO also handle isLoading
    // const errorMessage = useSelector<RootState, UrlToMusicState["error"]>(
    //     (state: RootState) => state.urlToMusic.error
    // );
    // TODO migrate all state to redux
    // const [url, setUrl] = React.useState("");
    const handleUrlChange = (event: React.ChangeEvent<HTMLInputElement>) => {
        dispatch(setUrl(event.target.value));
    };
    const dispatchGetInfo = () => {
        dispatch(getInfo());
    };
    const year = new Date().getFullYear();
    const [musicData, setMusicData] = React.useState({
        title: { value: "", error: false },
        artist: { value: "", error: false },
        album: { value: `Songs from ${year}`, error: false },
    });
    const handleMusicDataChange = (
        event: React.ChangeEvent<HTMLInputElement>
    ) => {
        setMusicData({
            ...musicData,
            [event.target.name]: {
                value: event.target.value,
                error: false,
            },
        });
    };
    const getMusic = () => {
        if (musicData.title.value === "") {
            setMusicData({
                ...musicData,
                title: {
                    value: musicData.title.value,
                    error: true,
                },
            });
        } else {
            dispatch(logInfo("NYI1" + JSON.stringify(musicData)));
        }
    };
    return (
        <Card className="card-dashboard-height">
            <CardContent>
                <form>
                    <TextField
                        label="URL"
                        name="url"
                        fullWidth={true}
                        value={url}
                        error={Boolean(errorMessage)}
                        helperText={
                            Boolean(errorMessage) ? errorMessage : false
                        }
                        onChange={handleUrlChange}
                    />
                    <Button color="primary" onClick={dispatchGetInfo}>
                        Get Info
                    </Button>
                </form>
                {/* TODO validate required */}
                <form>
                    <TextField
                        label="Title"
                        name="title"
                        fullWidth={true}
                        value={title}
                        error={musicData.title.error}
                        helperText={musicData.title.error ? "required" : false}
                        onChange={handleMusicDataChange}
                    />
                    <TextField
                        label="Artist"
                        name="artist"
                        fullWidth={true}
                        value={musicData.artist.value}
                        error={musicData.artist.error}
                        helperText={musicData.artist.error ? "required" : false}
                        onChange={handleMusicDataChange}
                    />
                    <TextField
                        label="Album"
                        name="album"
                        fullWidth={true}
                        value={musicData.album.value}
                        error={musicData.album.error}
                        helperText={musicData.album.error ? "required" : false}
                        onChange={handleMusicDataChange}
                    />
                    <Button color="primary" onClick={getMusic}>
                        Get Music
                    </Button>
                </form>
            </CardContent>
        </Card>
    );
};

export default UrlToMusic;
