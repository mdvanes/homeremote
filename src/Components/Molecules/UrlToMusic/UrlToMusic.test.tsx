import React from "react";
import { fireEvent, waitFor } from "@testing-library/react";
import UrlToMusic from "./UrlToMusic";
import urlToMusicReducer, { initialState } from "./urlToMusicSlice";
import { RootState } from "../../../Reducers";
import { renderWithProviders } from "../../../testHelpers";

const fetchSpy = jest.spyOn(window, "fetch");

type MockRootState = Pick<RootState, "urlToMusic">;

const mockRootState: MockRootState = {
    urlToMusic: initialState,
};

const renderUrlToMusic = (initialState: MockRootState) =>
    renderWithProviders(<UrlToMusic />, {
        initialState,
        reducers: {
            urlToMusic: urlToMusicReducer,
        },
    });

describe("UrlToMusic", () => {
    beforeEach(() => {
        const mockResponse: Partial<Response> = {
            ok: true,
            json: () =>
                Promise.resolve({
                    title: "Some Title",
                    artist: "Some Artist",
                }),
        };
        fetchSpy.mockResolvedValueOnce(mockResponse as Response);
    });

    afterEach(() => {
        fetchSpy.mockClear();
    });

    it("looks like a form", () => {
        const { asFragment } = renderUrlToMusic({
            ...mockRootState,
            urlToMusic: {
                ...initialState,
                form: {
                    ...initialState.form,
                    album: {
                        ...initialState.form.album,
                        value: "Mock Initial Album Name",
                    },
                },
            },
        });

        expect(asFragment()).toMatchSnapshot();
    });

    it("can submit the getinfo form", async () => {
        const { getByTestId } = renderUrlToMusic(mockRootState);

        expect(getByTestId("title").querySelector("input")).toHaveDisplayValue(
            ""
        );
        const urlInput = getByTestId("url").querySelector("input");
        if (urlInput) {
            fireEvent.change(urlInput, { target: { value: "Some URL" } });
        }
        const getInfoButton = getByTestId("get-info");
        fireEvent.click(getInfoButton);
        expect(fetchSpy).toHaveBeenCalledTimes(1);
        await waitFor(() =>
            expect(
                getByTestId("title").querySelector("input")
            ).toHaveDisplayValue("Some Title")
        );
        expect(getByTestId("artist").querySelector("input")).toHaveDisplayValue(
            "Some Artist"
        );
        expect(fetchSpy).toHaveBeenCalledTimes(1);
    });

    it("can submit the getmusic form", async () => {
        const mockResponse: Partial<Response> = {
            ok: true,
            json: () =>
                Promise.resolve({
                    path: "Some Path",
                    fileName: "Some FileName",
                }),
        };
        fetchSpy.mockReset().mockResolvedValueOnce(mockResponse as Response);

        const { getByTestId, getByText } = renderUrlToMusic(mockRootState);

        const urlInput = getByTestId("url").querySelector("input");
        if (urlInput) {
            fireEvent.change(urlInput, { target: { value: "Some URL" } });
        }
        const titleInput = getByTestId("title").querySelector("input");
        if (titleInput) {
            fireEvent.change(titleInput, { target: { value: "Some Title" } });
        }
        const artistInput = getByTestId("artist").querySelector("input");
        if (artistInput) {
            fireEvent.change(artistInput, { target: { value: "Some Artist" } });
        }
        const getMusicButton = getByTestId("get-music");
        fireEvent.click(getMusicButton);
        await waitFor(() =>
            expect(getByText(/Result in Some Path/)).toBeInTheDocument()
        );
        expect(fetchSpy).toHaveBeenCalledTimes(1);
    });
});
