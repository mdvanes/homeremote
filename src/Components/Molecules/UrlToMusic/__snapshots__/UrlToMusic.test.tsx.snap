// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UrlToMusic looks like a form 1`] = `
<DocumentFragment>
  <div
    class="MuiPaper-root MuiCard-root card-dashboard-height MuiPaper-elevation1 MuiPaper-rounded"
  >
    <div
      class="MuiCardContent-root"
    >
      <form>
        <div
          class="MuiFormControl-root MuiTextField-root MuiFormControl-fullWidth"
          data-testid="url"
        >
          <label
            class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated"
            data-shrink="false"
          >
            URL
          </label>
          <div
            class="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-fullWidth MuiInput-fullWidth MuiInputBase-formControl MuiInput-formControl"
          >
            <input
              aria-invalid="false"
              class="MuiInputBase-input MuiInput-input"
              name="url"
              type="text"
              value=""
            />
          </div>
        </div>
        <button
          class="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-textPrimary"
          data-testid="get-info"
          tabindex="0"
          type="button"
        >
          <span
            class="MuiButton-label"
          >
            Get Info
          </span>
          <span
            class="MuiTouchRipple-root"
          />
        </button>
      </form>
      <form>
        <div
          class="MuiFormControl-root MuiTextField-root MuiFormControl-fullWidth"
          data-testid="title"
        >
          <label
            class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated"
            data-shrink="false"
          >
            Title
          </label>
          <div
            class="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-fullWidth MuiInput-fullWidth MuiInputBase-formControl MuiInput-formControl"
          >
            <input
              aria-invalid="false"
              class="MuiInputBase-input MuiInput-input"
              name="title"
              type="text"
              value=""
            />
          </div>
        </div>
        <div
          class="MuiFormControl-root MuiTextField-root MuiFormControl-fullWidth"
          data-testid="artist"
        >
          <label
            class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated"
            data-shrink="false"
          >
            Artist
          </label>
          <div
            class="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-fullWidth MuiInput-fullWidth MuiInputBase-formControl MuiInput-formControl"
          >
            <input
              aria-invalid="false"
              class="MuiInputBase-input MuiInput-input"
              name="artist"
              type="text"
              value=""
            />
          </div>
        </div>
        <div
          class="MuiFormControl-root MuiTextField-root MuiFormControl-fullWidth"
        >
          <label
            class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiFormLabel-filled"
            data-shrink="true"
          >
            Album
          </label>
          <div
            class="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-fullWidth MuiInput-fullWidth MuiInputBase-formControl MuiInput-formControl"
          >
            <input
              aria-invalid="false"
              class="MuiInputBase-input MuiInput-input"
              name="album"
              type="text"
              value="Mock Initial Album Name"
            />
          </div>
        </div>
        <button
          class="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-textPrimary"
          data-testid="get-music"
          tabindex="0"
          type="button"
        >
          <span
            class="MuiButton-label"
          >
            Get Music
          </span>
          <span
            class="MuiTouchRipple-root"
          />
        </button>
      </form>
    </div>
  </div>
</DocumentFragment>
`;
